### PR TITLE
improve fluent interface on PoM, tests

### PR DESCRIPTION
**Pull request by ben-c-at-moz**
* Improve the POMs to be more amenable to the Fluent Interface model
* Refactor tests to use chained methods
* Fix return type typos in Browser Object Model

### CODE REVIEW

File: modules/browser_object_navigation.py [hunk 1]
```diff
@@ -74,18 +74,27 @@ def resume_content_context(self):
...
 
-    def get_awesome_bar(self) -> WebElement:
+    def expect(self, condition) -> Page:
+        self.wait.until(condition)
+        return self
+
+    def expect_in_content(self, condition) -> Page:
+        with self.driver.context(self.driver.CONTEXT_CONTENT):
+            self.expect(condition)
+        return self
```
[the remark this answers is not in the document]
Comment (ben-c-at-moz) [reply modules/browser_object_navigation.py:84]: Yeah, trying to think about how exactly that would work, but you're absolutely right.

File: modules/page_object_about_logins.py [hunk 2]
```diff
@@ -59,11 +59,16 @@ def login_item_save_changes_button(self) -> WebElement:
             By.CLASS_NAME, "save-changes-button"
         )
 
-    def create_new_login(self, form_info: dict):
+    def click_add_login_button(self) -> Page:
+        self.add_login_button().click()
```
[the remark this answers is not in the document]
Comment (ben-c-at-moz) [reply modules/page_object_about_logins.py:63]: also a good call

File: modules/browser_object_navigation.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ben-c-at-moz) [reply modules/browser_object_navigation.py]: I think I got it

File: tests/address_bar_and_search/test_search_modes_for_sites.py [hunk 3]
```diff
@@ -14,8 +14,9 @@ def test_search_modes_for_sites(driver: Firefox, site: str):
...
-    nav.type_in_awesome_bar("soccer" + Keys.RETURN)
-    with driver.context(driver.CONTEXT_CONTENT):
-        nav.wait.until(EC.url_contains(site.lower()))
-    nav.awesome_bar.clear()
+    (
+        nav.set_search_mode_via_awesome_bar(site)
+        .type_in_awesome_bar("soccer" + Keys.RETURN)
+        .expect_in_content(EC.url_contains(site.lower()))
+        .awesome_bar.clear()
+    )
```
[the remark this answers is not in the document]
Comment (jrbenny35) [reply tests/address_bar_and_search/test_search_modes_for_sites.py]: Ah I gotcha. I am more used to the latter but wouldn't be opposed to yours looking different. It's more so your choice!

File: modules/page_base.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+
+class BasePage(Page):
+    def __init__(self, driver, **kwargs):
+        super().__init__(driver, **kwargs)
+        self.utils = PomUtils(self.driver)
+
+    def expect(self, condition) -> Page:
+        self.wait.until(condition)
+        return self
```
Comment (jrbenny35) [inline modules/page_base.py:13]: Personally love this!

File: modules/browser_object_navigation.py [hunk 5]
```diff
@@ -88,6 +86,11 @@ def get_awesome_bar(self) -> Page:
         self.awesome_bar = self.driver.find_element(*self._awesome_bar)
         return self
 
+    def clear_awesome_bar(self) -> Page:
+        self.get_awesome_bar()
+        self.awesome_bar.clear()
```
Comment (jrbenny35) [inline modules/browser_object_navigation.py:91]: This looks a little weird, you're not returning the awesome bar from `get_awesome_bar` but you can clear it?

**Review by jrbenny35 — CHANGES_REQUESTED** [review]
Lots of nice changes, just had some follow up thoughts

**Review by jrbenny35 — APPROVED** [review]
/giphy "just gonna send it", Nice work!

**Pull request merged**